src/utils.rs

```diff
@@ -15,3 +25,43 @@ impl AsUriExt for Url {
         Uri::try_from(self.to_string()).unwrap()
     }
 }
+
+/// Validates guardrails on request.
+pub fn validate_guardrails(
+    request_detectors: &HashMap<String, DetectorParams>,
+    orchestrator_detectors: &HashMap<String, DetectorConfig>,
+    allowed_detector_types: Vec<DetectorType>,
+    allows_whole_doc_chunker: bool,
+) -> Result<(), Error> {
+    let whole_doc_chunker_id = DEFAULT_CHUNKER_ID;
+    let detector_ids = request_detectors.keys();
+    for detector_id in detector_ids {
+        // validate detectors
+        match orchestrator_detectors.get(detector_id) {
+            Some(detector_config) => {
+                if !allowed_detector_types.contains(&detector_config.r#type) {
+                    let error = Error::Validation(format!(
+                        "{}: detector is not supported on this endpoint",
+                        detector_id
+                    ));
+                    error!(%error, "invalid detector on request");
```

Comment:
nit: error Display impl already provides descriptive message
```suggestion
                    error!("{error}");
```